src/scripts/gdb/strubtest.py

```diff
@@ -0,0 +1,131 @@
+# (C) 2025 Jack Lloyd
+#     2025 René Meusel, Rohde & Schwarz Cybersecurity
+#
+# Botan is released under the Simplified BSD License (see license.txt)
+#
+
+"""
+User-defined GDB CLI command to help in testing GCC's stack scrubbing
+annotations. See also src/scripts/test_strubbed_symbols.py for details.
+
+This can be used in interactive GDB sessions for debugging and development.
+
+   $> gdb -x src/scripts/gdb/strubtest.py ./botan
+   (gdb) strubtest Botan::SHA_256::compress_digest
+   strubtest setup done
+   (gdb) run hash --algo=SHA-256 readme.rst
+   Success: stackframe of Botan::SHA_256::compress_digest contains 10176 zero bytes after invocation
+   ECD8814EDC180601831FDAD7DEDCF24D57F9F002295346E7D558C941AD318F8E readme.rst
+"""
+
+import gdb
+
+def stack_pointer(frame):
+    # Currently this will likely work on x86_64 only.
+    # TODO: find a better way to do that!
+    return frame.read_register('rsp')
+
+def base_pointer(frame):
+    # Currently this will likely work on x86_64 only.
+    # TODO: find a better way to do that!
+    return frame.read_register('rbp')
+
```

Comment:
`$fp` maybe? (Haven't tested this beyond checking that gdb knows about `$fp` on x86)

Reply:
Works on my machine as well (which is also x86).